lib/libsmu.c

```diff
@@ -126,9 +131,7 @@ static smu_return_val smu_init_parse(smu_obj_t* obj) {
         return SMU_Return_RWError;
 
     obj->codename = atoi(rd_buf);
-
-    if (obj->codename <= CODENAME_UNDEFINED ||
-        obj->codename >= CODENAME_COUNT)
+    if (obj->codename <= CODENAME_UNDEFINED || obj->codename >= CODENAME_COUNT)
```

Comment:
Logic is now broken as codename is now a string so smu_init will fail now. Maybe just revert that change and add another sysfs file like codename_str if you really want that functionality?